sips/sip-02x/sip-02x-non-sequential-multisig-transactions.md

```diff
@@ -0,0 +1,137 @@
+# Preamble
+
+SIP Number: 02x
+
+Title: Non-sequential Multisig Transactions
+
+Author: Jeff Bencin <jbencin@hiro.so>
+
+Consideration: Technical
+
+Type: Consensus
+
+Status: Draft
+
+Created: 2023-08-30
+
+License: BSD 2-Clause
+
+Sign-off:
+
+Discussions-To: https://github.com/stacksgov/sips
+
+# Abstract
+
+This SIP proposes a new multisig transaction format which is intended to be easier to use than the current format described in SIP-005.
+It does not remove support for the current format, rather it is intended to co-exist with the old format and give users a choice of which format to use.
+
+The issue with the current format is that it establishes a signer order when funds are sent to multisig account address, and requires signers to sign in the same order to spend the funds.
+In practice, the current format has proven proven difficult for developers to understand and implement, as evidenced by the lack of Stacks multisig implementations today.
+
+This new format intends to simplify the signing algorithm and remove the requirement for in-order signing, without comprimising on security or increasing transaction size.
+It is expected that this will lead to better wallet support for Stacks multisig transactions.
+
+# Introduction
+
+Currently, a multisig transaction requires the first signer to sign the transaction itself, and following signers to sign the signature of the previous signer.
+For a transaction with *n* signers, the final signature is generated in the following way:
+
+```
+signature_n(...(signature_2(signature_1(tx))))
+```
+
+There are a few drawbacks to doing it this way:
+
+- The order in which the signers must sign is fixed as soon as funds are send to a multisig account, which limits flexibility when creating a spending transaction from a multisig account
+- The process of signing a transaction requires each signer to validate the entire signature chain before signing, in order to make sure it matches the transaction, leading to `O(n^2)` signing times
+- This does not reduce the size of a transaction, as each intermediate signature must still be included
+- The algorithm for doing this is complex, and several developers have a hard time understanding and implementing it correctly
+
+This document proposes having each signer sign the transaction directly:
+
+```
+signature_1(tx), signature_2(tx), ..., signature_n(tx)
+```
+
+This would address all of the concerns listed above, and would not increase transaction size or make it easier to forge a signature
```

Comment:
I would add something like:
Imagine a DAO that has 5 management team members. They decide to make a 3 out of 5 multisig on Stacks.
With the current version, signing a transaction in the creation order is required, i.e., after the 3rd owner only the 4th and 5th can sign, after the 5th owner signed - no one can add another signature.
It makes usage of the multisig painful for a lot of scenarios, for example:
- It is impossible to start signing a transaction from the 4th and 5th owners since they won't have enough owners after them to finalize the transaction
- If the 1st owner initiated a transaction, the 4th owner signed, the 2nd and 3rd owners can't sign this transaction anymore, so only the 5th owner can finalize it, the case of 2nd or 3rd owners being online and 5th not - they need to wait until the last owner makes this signature.
- If the 3rd owner initiated and signed a transaction, he needs the 4th and 5th owners' signatures in order to finalize it. 1st and 2nd owners can't participate until they make a new signing flow for that transaction.
In short, for small multisig wallets it seems to be not a blocking problem, but for multisig wallets with 4 or more owners, it becomes really hard to initiate transactions and get signatures in place.
@jo-tm @jbencin

Reply:
@fess-v This is a good example, it shows that when you establish signer order by creating a multisig address, not all signers are equal. Those near the top of the list have much more flexibility than those near the bottom

Would you please add this to the draft SIP? While you're at it, add your name and email to the authors list, you've done a lot of work on this

Reply:
Sure! will do that soon, thank you @jbencin